emissary-core/src/sam/session.rs

```diff
@@ -743,7 +754,16 @@ impl<R: Runtime> SamSession<R> {
 
         match self.destination.query_lease_set(&destination_id) {
             LeaseSetStatus::Found => {
-                let datagram = self.datagram_manager.make_datagram(protocol, datagram);
+                let datagram = match protocol {
+                    Protocol::Anonymous => self.datagram_manager.make_anonymous(datagram),
+                    Protocol::Datagram => self.datagram_manager.make_datagram(datagram),
+                    Protocol::Datagram2 => self.datagram_manager.make_datagram2(
+                        datagram,
+                        &Sha256::new().update(destination.as_ref()).finalize(),
+                        None, // TODO datagram2: where should I get the options?
```

Comment:
There's support for parsing [options](https://github.com/altonen/emissary/blob/c4a1c849ebfceba892adce53f512f1f099721de2/emissary-core/src/sam/parser.rs#L774-L776) but they're currently ignored by `SamServer` https://github.com/altonen/emissary/blob/c4a1c849ebfceba892adce53f512f1f099721de2/emissary-core/src/sam/mod.rs#L375-L380

so a new `options` field needs to be added to `SamSessionCommand::SendDatagram`